torchtitan/experiments/gpt_oss/model/model.py

```diff
@@ -0,0 +1,407 @@
+        self.wk = nn.Linear(
+            model_args.dim,
+            model_args.n_kv_heads * model_args.head_dim,
+            bias=True,
+        )
+        self.wv = nn.Linear(
+            model_args.dim,
+            model_args.n_kv_heads * model_args.head_dim,
+            bias=True,
+        )
+        self.wo = nn.Linear(
+            model_args.n_heads * model_args.head_dim,
+            model_args.dim,
+            bias=True,
+        )
+        self.sinks = nn.Parameter(torch.empty(model_args.n_heads))
+
+        self.use_flex_attn = getattr(model_args, "use_flex_attn", False)
+
+        if self.use_flex_attn:
+            self.inner_attention = FlexAttentionWrapper()
+        else:
+            raise ValueError("Gpt-oss model only supports FlexAttention!")
+
+    def init_weights(self, init_std: float):
+        linear_list = [
+            self.wq,
+            self.wk,
+            self.wv,
+        ]
+
+        nn.init.trunc_normal_(self.sinks, mean=0.0, std=init_std)
+        for linear in linear_list:
+            nn.init.trunc_normal_(linear.weight, mean=0.0, std=init_std)
+            nn.init.trunc_normal_(linear.bias, mean=0.0, std=init_std)
+        nn.init.trunc_normal_(self.wo.weight, mean=0.0, std=init_std)
+        nn.init.trunc_normal_(self.wo.bias, mean=0.0, std=init_std)
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        rope_cache: torch.Tensor,
+        attention_masks: AttentionMasksType | None,
+    ):
+        """
+        Forward pass for the Multi-Head Latent Attention (MLA) Layer.
+
+        Args:
+            x (torch.Tensor): Input tensor of shape (batch_size, seq_len, dim).
+            rope_cache (torch.Tensor): Precomputed cosine and sine frequencies for rope embedding.
+
+        Returns:
+            torch.Tensor: Output tensor with the same shape as the input.
+        """
+        bsz, seqlen, _ = x.size()
+        hidden_shape = (bsz, seqlen, -1, self.head_dim)
+
+        q = self.wq(x).view(hidden_shape)
+        k = self.wk(x).view(hidden_shape)
+        v = self.wv(x).view(hidden_shape)
+
+        q, k = apply_rotary_emb(q, k, rope_cache)
+
+        # repeat k/v heads if n_kv_heads < n_heads
+        keys = repeat_kv(k, self.n_rep)
+        values = repeat_kv(v, self.n_rep)
+
+        xq = q.transpose(1, 2).contiguous()
+        xk = keys.transpose(1, 2).contiguous()
+        xv = values.transpose(1, 2).contiguous()
+
+        if self.use_flex_attn:
+            assert isinstance(attention_masks, BlockMask), attention_masks
+            output, lse = self.inner_attention(
+                xq, xk, xv, block_mask=attention_masks, scale=None, return_lse=True
+            )
+
+            # Apply attention sink rescaling: rescale by σ(lse - w[h])
+            # This is mathematically equivalent to concatenating learnable sink weights
+            sink_scale = torch.sigmoid(lse - self.sinks.view(1, -1, 1)).unsqueeze(-1)
+            output = output * sink_scale.to(output.dtype)
```

Comment:
What if `use_flex_attn` is False?  output will be undefined. If we only support FlexAttention case, we should error out during the initialization if `use_flex_attn` False.